File: example/javascriptlib/module/1-common-config/foo/custom-src/foo2.ts
import * as fs from 'fs/promises';
import FooA from "@generated/foo-A";
import FooB from "@generated/foo-B";
import FooC from "@generated/foo-C";

const Resources: string = process.env.RESOURCES || "@foo/resources" // `RESOURCES` is generated on bundle
const CustomResources: string = process.env.CUSTOMRESOURCES || "@foo/custom-resources"

const MyResource = require.resolve(`${Resources}/MyResource.txt`);
const MyOtherResource = require.resolve(`${CustomResources}/MyOtherResource.txt`);


export default class Foo2 {
    static value: string = "hello2"

    static async resourceText(filePath): Promise<string> {
        try {
            return await fs.readFile(filePath, 'utf8');
        } catch (err) {
            console.error('Error reading the file:', err);
            throw err;
        }
    }
}

(async function () {
    console.log(Foo2.value)
    console.log(FooA.value)
    console.log(FooB.value)
    console.log(FooC.value)
    console.log(process.env.MY_CUSTOM_ENV)
    console.log("MyResource: " + await Foo2.resourceText(MyResource))
    console.log("MyOtherResource: " + await Foo2.resourceText(MyOtherResource))
})()

File: example/kotlinlib/module/2-common-config/custom-src/Foo2.kt
package foo

import kotlinx.html.h1
import kotlinx.html.stream.createHTML
import java.io.IOException

fun main(args: Array<String>) {
    println("Foo2.value: ${Foo2.VALUE}")
    println("Foo.value: ${Foo.VALUE}")

    println("FooA.value: ${FooA.VALUE}")
    println("FooB.value: ${FooB.VALUE}")
    println("FooC.value: ${FooC.VALUE}")

    println("MyResource: ${readResource("MyResource.txt")}")
    println("MyOtherResource: ${readResource("MyOtherResource.txt")}")

    val properties = System.getProperties()
    println("my.custom.property: ${properties.getProperty("my.custom.property")}")

    val myCustomEnv = System.getenv("MY_CUSTOM_ENV")
    if (myCustomEnv != null) {
        println("MY_CUSTOM_ENV: $myCustomEnv")
    }
}

object Foo2 {
    val VALUE = createHTML(prettyPrint = false).h1 { text("hello2") }.toString()
}

private fun readResource(resourceName: String): String? = try {
    ::main.javaClass.classLoader
        .getResourceAsStream(resourceName)
        .readAllBytes()
        .toString(Charsets.UTF_8)
} catch (e: IOException) {
    null
}
